File: 백준/Silver/17413. 단어 뒤집기 2/단어 뒤집기 2.js
const S = require("fs").readFileSync(0).toString().trim().split("");
let dequeue = [];
let result = "";
S.forEach((el, i) => {
  if (el === "<") {
    result += dequeue.reverse().join("");
    dequeue = [];
    dequeue.push("<");
  } else if (el === ">") {
    dequeue.push(">");
    result += dequeue.join("");
    dequeue = [];
  } else if (dequeue[0] === "<") {
    dequeue.push(el);
  } else if (el === " ") {
    result += dequeue.reverse().join("") + " ";
    dequeue = [];
  } else if (el === el.toLowerCase()) {
    dequeue.push(el);
    if (i === S.length - 1) {
      result += dequeue.reverse().join("");
    }
  }
});
console.log(result);

File: 백준/Silver/17413. 단어 뒤집기 2/단어 뒤집기 2.py
S = input()

i=0
result=""
stack=[]
while i<len(S):
    if S[i]=="<":
        while S[i]!=">":
            result+=S[i]
            i+=1
        result+=S[i]
        i+=1
    elif S[i]==" ":
        result+=S[i]
        i+=1
    else:   
        while (i<len(S) and S[i]!="<") and (i<len(S) and S[i]!=" "):
            stack.append(S[i])
            i+=1
        stack.reverse()
        result+="".join(stack)
        stack=[]

print(result)
        